modules/network/samba.c

```diff
@@ -51,33 +51,36 @@ void
 scan_samba_usershares(void)
 {
     FILE *usershare_list;
-    
-    if ((usershare_list = popen("net usershare list", "r"))) {
-        char buffer[512];
-        
+    gchar *out, *err;
+    gboolean spawned;
+    int status;
+    gchar *p, *next_nl;
+    gchar *usershare, *cmdline;
+    gsize length;
+
+    spawned = g_spawn_command_line_sync("net usershare list",
+            &out, &err, &status, NULL);
```

Comment:
If NULL is passed, the following message appears in Terminal when Shared Directories is selected:
```
net usershare: cannot open usershare directory /var/lib/samba/usershares. Error No such file or directory
Please ask your system administrator to enable user sharing.
```
I don't know, but I think it could still be optional and just forward stderr to the application's stderr if it is NULL. 
When it is captured in a string, the message does not appear in Terminal.